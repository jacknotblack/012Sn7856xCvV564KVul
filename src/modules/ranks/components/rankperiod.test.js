import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RankPeriod from "./rankperiod";

describe("A suite", () => {
  let wrapper;
  const props = {
    selectedRankPeriod: "daily",
    onRankPeriodClick: jest.fn(),
    periods: ["daily", "weekly", "monthly"]
  };

  beforeEach(() => {
    wrapper = shallow(<RankPeriod {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render 3 period options", () => {
    expect(wrapper.find(".periodItem")).toHaveLength(props.periods.length);
  });

  it("should render period options with right names", () => {
    expect(wrapper.find(".periodItem").map(n => n.text())).toEqual(
      props.periods
    );
  });

  it("should call onRankPeriodClick if onClick", () => {
    const tab = wrapper.find(".periodItem");
    tab.map(t => t.props().onClick());
    expect(props.onRankPeriodClick.mock.calls.length).toBe(tab.length);
    expect(props.onRankPeriodClick.mock.calls.map(c => c[0])).toEqual(
      props.periods
    );
  });
});
