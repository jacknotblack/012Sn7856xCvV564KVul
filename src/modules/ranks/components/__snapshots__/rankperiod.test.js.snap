// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite should match snapshot 1`] = `
<div>
  Selected:
  daily
  <div
    className="periodItem"
    key="daily"
    onClick={[Function]}
    role="presentation"
  >
    daily
  </div>
  <div
    className="periodItem"
    key="weekly"
    onClick={[Function]}
    role="presentation"
  >
    weekly
  </div>
  <div
    className="periodItem"
    key="monthly"
    onClick={[Function]}
    role="presentation"
  >
    monthly
  </div>
</div>
`;
